core/src/main/java/cz/o2/proxima/repository/AttributeValueDescriptor.java

```diff
@@ -0,0 +1,999 @@
+
+  /** Return {@link Descriptor} for give attribute value */
+  @Getter private final Descriptor<T> schemaDescriptor;
+
+  private AttributeValueDescriptor(Descriptor<T> schemaDescriptor) {
+    this.schemaDescriptor = schemaDescriptor;
+  }
+
+  /**
+   * Create {@link AttributeValueDescriptor} with Primitive type.
+   *
+   * @param descriptor primitive type descriptor
+   * @param <T> attribute type
+   * @return Attribute value descriptor
+   */
+  public static <T> AttributeValueDescriptor<T> of(PrimitiveTypeDescriptor<T> descriptor) {
+    return of(descriptor.toDescriptor());
+  }
+
+  /**
+   * Create {@link AttributeValueDescriptor} with Array type
+   *
+   * @param descriptor Array type descriptor
+   * @param <T> attribute type
+   * @return Attribute value descriptor
+   */
+  public static <T> AttributeValueDescriptor<T> of(ArrayTypeDescriptor<T> descriptor) {
+    return of(descriptor.toDescriptor());
+  }
+
+  /**
+   * Create {@link AttributeValueDescriptor} with Structure type
+   *
+   * @param descriptor Structure type descriptor
+   * @param <T> attribute type
+   * @return attribute value descriptor
+   */
+  public static <T> AttributeValueDescriptor<T> of(StructureTypeDescriptor<T> descriptor) {
+    return of(descriptor.toDescriptor());
+  }
+
+  /**
+   * Create {@link AttributeValueDescriptor} from generic {@link Descriptor}.
+   *
+   * @param descriptor Generic descriptor
+   * @param <T> attribute type
+   * @return attribute value descriptor
+   */
+  public static <T> AttributeValueDescriptor<T> of(Descriptor<T> descriptor) {
+    return new AttributeValueDescriptor<>(descriptor);
+  }
+
+  /** Possible attribute and field types */
+  public enum AttributeValueType {
+    STRUCTURE,
+    ARRAY,
+    ENUM,
+    BYTE,
+    INT,
+    LONG,
+    DOUBLE,
+    FLOAT,
+    STRING,
+    BOOLEAN,
+  }
+
+  /**
+   * Default value factory
+   *
+   * @param <T> attribute type
+   */
+  @FunctionalInterface
+  public interface DefaultValueFactory<T> extends Serializable {
+
+    /** Provide default value */
+    T apply();
+  }
+
+  /**
+   * Interface provides default value for attribute or other structure.
+   *
+   * @param <T> value type
+   */
+  public interface WithDefaultValue<T> {
+
+    /** Return {@code true} when default value can be provided. */
+    default boolean hasDefaultValue() {
+      return getDefaultValue() != null;
+    }
+
+    /** Return default value or null */
+    @Nullable
+    T getDefaultValue();
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    AttributeValueDescriptor<?> that = (AttributeValueDescriptor<?>) o;
+    return schemaDescriptor.equals(that.schemaDescriptor);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(schemaDescriptor);
+  }
+
+  /** Descriptors for types. */
+  public static class Descriptors {
```

Comment:
We have way to many descriptors throughout the code base, so we should be more specific with the naming

`Descriptor` -> `SchemaDescriptor`
`Descriptors` -> `SchemaDescriptors`

Maybe would you consider moving this to a top-level class?

Reply:
Renamed to:
Descriptors -> SchemaDescriptors
Descriptor -> SchemaTypeDescriptor

also moved to top-level class and move to `cz.o2.proxima.scheme` package